Staging feature/login

DaikoneKisu opened the pull request.

DaikoneKisu committed:
feat: added SignInLogInSwitchButton component and some tests :D

Noralgorithm committed:
chore: tailwind config

DaikoneKisu committed:
Merge remote-tracking branch 'remotes/origin/feature/sign-in-log-in-switch-button' into feature/login

Noralgorithm committed:
feat: layout and img sources

DaikoneKisu committed:
Merge branch 'main' into feature/login

DaikoneKisu committed:
feat: add figma text styles to app.css

DaikoneKisu committed:
chore: add new font families to tailwind theme

DaikoneKisu committed:
fix: resolve a conflict in tailwind.config.js

DaikoneKisu committed:
Merge remote-tracking branch 'remotes/origin/feature/sign-up-and-login-layout' into feature/login

DaikoneKisu committed:
feat: add key.svg :D

DaikoneKisu committed:
fix: modify +layout.svelte to be up to date with its counterpart in branch 'featre/sign-up'

DaikoneKisu committed:
Merge branch 'feature/sign-up-and-login-layout' into feature/login

DaikoneKisu committed:
fix: modify +layout.svelte because h1 and h2 elements in /sign-up had different classes than those in /login (and they shouldn't), and adapt the classes to not be redundant with app.css

DaikoneKisu committed:
feat: add Login Form component to /login

DaikoneKisu committed:
test: add tests to Login Form Component in /login

Noralgorithm committed:
feat: create password input component

Noralgorithm committed:
test: test password input component

Noralgorithm committed:
fix: delete index page password input test

Noralgorithm committed:
fix: set classname and label as optional props

Noralgorithm committed:
refactor: put password input component and test into components folder

Noralgorithm committed:
fix: set value variable as prop

Noralgorithm committed:
fix: add placeholder prop

DaikoneKisu committed:
feat: add name prop to InputWithIcon component andadapt tests to send it

DaikoneKisu committed:
refactor: move src/routes/+layout.svelte file to src/routes/(auth)/+layout.svelte and files in src/routes/login to src/routes/(auth)/login

DaikoneKisu committed:
feat: add token session cookie setting functionality

add SessionData interface and isSessionData and getSessionData functions
add method POST to login form so it can be retrieved in a server action
add server action to get session data and set session cookie
add tests related to SessionData related stuff (interface and functions)

DaikoneKisu committed:
refactor: create components folder and move src/lib/input-with-icon folder and its files to src/lib/components/input-with-icon

DaikoneKisu committed:
feat: add name prop to InputWithIcon.svelte, delete input of type password from InputWithIcon.svelte, adapt tests in input-with-icon.test.ts and change InputWithIcon.svelte formatting

DaikoneKisu committed:
style: change white-spacing and format of PasswordInput.svelte and password-input.test.ts files

DaikoneKisu committed:
test: add missing prop value to PasswordInput component rendered in password-input.test.ts

DaikoneKisu committed:
feat: add appropriate alt attribute value to eye and eyeInvisible images in PasswordInput component

DaikoneKisu committed:
fix: re-improve input with icon styles 'cuz improvement got deleted in a previous commit

DaikoneKisu committed:
Merge branch 'feature/input-with-icon' into feature/login

DaikoneKisu committed:
chore: move src/lib/sign-in-log-in-switch-button folder and files to src/lib/components/sign-in-log-in-switch-button

DaikoneKisu committed:
Merge branch 'feature/sign-up-log-in-switch-button' into feature/login

DaikoneKisu committed:
fix: solve 'failed to load url' bugs caused by previous commits moving components from lib to lib/components

DaikoneKisu committed:
Merge branch 'feature/password-input' into feature/login

DaikoneKisu committed:
feat: add name prop to PasswordInput component

DaikoneKisu committed:
Merge branch 'feature/password-input' into feature/login

DaikoneKisu committed:
refactor: do changes in Form login component

use PasswordInput component in Form login component instead of InputWithIcon of type password
remove aria-label attribute from Form login component submit button 'cuz it's not necessary
adapt Form login component tests to changes made to the submit button

DaikoneKisu committed:
refactor: change the way http status codes are managed

create http-status-codes.ts with an enum of http status codes (not ours, give credit to creator)
change get-session-data.ts to use StatusCodes enum
change login's +page.server.ts to use StatusCodes enum

DaikoneKisu committed:
feat: add required prop to InputWithIcon component and tests

DaikoneKisu committed:
test: add test to check if input is not required if specified

DaikoneKisu committed:
Merge branch 'feature/input-with-icon' into feature/login

DaikoneKisu committed:
build: add yup dependency

DaikoneKisu committed:
chore: allow for dark mode to be toggeable

DaikoneKisu committed:
feat: mock sign-up page

DaikoneKisu committed:
Merge branch 'feature/input' into feature/login

DaikoneKisu committed:
refactor: create schema to validate user credentials and an interface to represent a login fail data

DaikoneKisu committed:
refactor: change isSessionData impl so that it uses yup, also define a sessionDataSchema

DaikoneKisu committed:
test: change some tests descriptions to better resemble what they actually test

DaikoneKisu committed:
feat!: add login form fields validations and display errors, also redirect on successful login

BREAKING CHANGE: Form component now has props email and password with no default values

DaikoneKisu committed:
feat: improve login redirect on success to allow redirection to any of the sites routes

Noralgorithm commented on src/routes/(auth)/login/Form.svelte:
Falta formatear el código, hay errores de indentación

Noralgorithm commented on src/routes/(auth)/login/+page.server.ts:
```diff
@@ -0,0 +1,44 @@
+    if (cookies.get('session')) {
+      const redirectTo = url.searchParams.get('redirectTo')
+      if (redirectTo && redirectTo !== '/')
+        throw redirect(StatusCodes.SEE_OTHER, `/${redirectTo}`)
```
Acá el "/" ya viene en el URL param, por lo que hay quitarlo de acá para evitar que cuando te redireccione se interprete como una URL absoluta. En su defecto, se podría quitar el "/" del valor almacenado en la URL y que siempre se la añada acá, ambas opciones solucionan el problema.
```suggestion
        throw redirect(StatusCodes.SEE_OTHER, `${redirectTo}`)
```

DaikoneKisu replied:
Listo

Noralgorithm commented on src/routes/(auth)/login/+page.server.ts:
Faltó formatear el código del archivo

Noralgorithm commented on src/routes/(auth)/login/+page.svelte:
Faltó formatear el código del archivo

Noralgorithm commented on src/routes/(auth)/login/login-fail-data.type.ts:
Faltó formatear el código del archivo

Noralgorithm commented on src/routes/(auth)/login/user-schema.ts:
Faltó formatear el código del archivo

Noralgorithm commented on src/routes/(authed)/+layout.server.ts:
Faltó formatear el código del archivo

Noralgorithm commented on tailwind.config.js:
Falta formatear el código, pero la configuración del DarkMode no se debería hacer en esta rama

DaikoneKisu replied:
Fuck Single Responsibility Principle

Noralgorithm commented on src/routes/(auth)/login/+page.svelte:
```diff
@@ -0,0 +1,111 @@
+		if (form?.reason) disabled = false
+	}
+
+	onMount(async () => {
+		await tick()
+		disabled = true
+	})
```
No entiendo para qué está esto

Hector1567XD replied:
La verdad yo tampoco, que diablos estan haciendo aqui 🤨

DaikoneKisu replied:
Yo menos, se corrige después

Hector1567XD commented on src/routes/(auth)/login/+page.server.ts:
```diff
@@ -0,0 +1,44 @@
+		const formData = await request.formData()
+    
+		try {
+      const session = await fetch('https://uchamba-backend-staging.1.us-1.fl0.io/auth/login', {
```
Ok pero
```suggestion
      const session = await fetch(`${BASE_URL}/auth/login`, {
```

DaikoneKisu replied:
xd, gracias

DaikoneKisu replied:
Lito

Noralgorithm committed:
feat: implement token hook, change redirect url and change app.css import to root layout

DaikoneKisu committed:
fix: add root routes layout and import app.css there

DaikoneKisu committed:
refactor: delete unnecessary app.css import

DaikoneKisu committed:
feat!: delete sign up feature traces

BREAKING CHANGE: sign up related stuff (routes, files, components) no longer exist

DaikoneKisu committed:
feat: use environment variables

DaikoneKisu committed:
refactor: tell git that line end char is lf

Paulaforero reviewed (changes requested):
Los inputs se ven muy grandes, al igual que los elementos en la página, los mensajes de aviso en los inputs deben estar alineados a la izquierda y tratar de que cuando aparezcan no se rueden los inputs ya que se ve algo raro. También mejorar el placeholder del input para dar un mejor ejemplo de como es el formato del correo UCAB.

DaikoneKisu committed:
feat: set up prettier, husky, lint-staged and git pre-commit format hook

Also, so many files changed due to the format hook lol

DaikoneKisu committed:
Merge branch 'feature/prettier' into feature/login, thus styling a lot of files due to git hook that runs prettier on commit

DaikoneKisu committed:
style: format all .css, .js, .ts and .svelte files

Noralgorithm reviewed (approved):
lgtm

Hector1567XD reviewed (approved):
Noc de Svelte, pero si se que estas bello <3

DaikoneKisu committed:
Merge branch 'staging' into feature/login

DaikoneKisu commented:
Se logró señores
